lib/ain-ocean/src/indexer/oracle.rs

```
@@ -39,11 +44,202 @@ impl Index for UpdateOracle {
+    // Convert Result to Option
+    let oracle_id = services
+        .oracle_token_currency
+        .by_key
+        .list(Some((token.to_string(), currency.to_string())))
+        .ok();
+    let mut oracle_token_currencies = Vec::new();
+
+    if let Some(mut oracle_iter) = oracle_id {
+        loop {
+            match oracle_iter.next() {
+                Some(Ok((_, value))) => {
+                    let oracle = services.oracle_token_currency.by_id.get(&value);
+                    if let Ok(Some(oracle)) = oracle {
+                        oracle_token_currencies.push(oracle);
+                    }
+                }
+
+                Some(Err(e)) => {
+                    println!("Error: {:?}", e);
+                    break;
+                }
+
+                None => break,
+            }
+        }
+    }
+
+    let result = oracle_token_currencies;
+    let mut aggregated = OraclePriceAggregatedAggregated {
+        amount: "0".to_string(),
+        weightage: 0,
+        oracles: OraclePriceAggregatedAggregatedOracles {
+            active: 0,
+            total: 0,
+        },
+    };
+
+    for oracle in result {
+        if oracle.weightage == 0 {
+            continue;
+        }
+
+        let key = (token.to_string(), currency.to_string(), oracle.oracle_id);
+        let feed_id = services.oracle_price_feed.by_key.get(&key);
+
+        let feeds = match feed_id {
+            Ok(feed_ids) => feed_ids.map_or_else(
+                || Ok(None),
+                |id| {
+                    services
+                        .oracle_price_feed
+                        .by_id
+                        .get(&id)
+                        .map(|data| Some(data))
+                },
+            ),
+            Err(err) => Err(err), // Pass along the error if there was one
+        };
+
+        let oracle_price_feed: Option<OraclePriceFeed> = match feeds {
+            Ok(Some(Some(feed))) => Some(feed),
+            _ => None,
+        };
```

Same as above, shouldn't silence the error